lib/sarif/gosec_sarif.rb

```diff
@@ -5,28 +5,37 @@ class GosecSarif < BaseSarif
     def initialize(scan_report)
       super(scan_report)
       @uri = GOSEC_URI
-      begin
-        json_obj = JSON.parse(scan_report.log(''))
-        issues = json_obj['Issues']
-        errors = json_obj['Golang errors']
-        parsed_errors = []
-        errors.each do |key|
-          key[1].each do |location|
-            location['uri'] = key[0]
-            parsed_errors << parse_error(location)
-          end
+      @issues = Set.new
+      @logs = parse_scan_report!(scan_report)
+    end
+
+    def parse_scan_report!(scan_report)
+      json_obj = JSON.parse(scan_report.log(''))
+      issues = json_obj['Issues']
+      errors = json_obj['Golang errors']
+      parsed_errors = []
+      errors.each do |key|
+        key[1].each do |location|
+          location['uri'] = key[0]
+          parsed_errors << parse_error(location)
         end
-        @logs = issues.concat parsed_errors
-      rescue JSON::ParserError
-        @logs = []
       end
+      parsed_errors.compact!
+      issues.concat parsed_errors
+    rescue JSON::ParserError
+      []
     end
 
     def parse_error(error)
       line = error['line'].to_i
       column = error['column'].to_i
       line = 1 if line.zero?
       column = 1 if column.zero?
+
+      id = error['error'] + error['uri'] + line.to_s
```

Comment:
Well, this could cause issues with edge cases.

Let's say
error 1 has uri "http://abcd.com/1", line 123
error 2 has uri "http://abcd.com/11", line 23

The code will consider these two different issues equivalent.

I think a simple fix is just update the code to 
`id = error['error'] + ' ' + error['uri'] + ' ' + line.to_s`